storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java

```diff
@@ -1183,11 +1188,12 @@ public void addNonCanonicalSidecar(final DataColumnSidecar sidecar) {
   }
 
   @Override
-  public void pruneAllSidecars(final UInt64 tillSlotInclusive) {
+  public void pruneAllSidecars(final UInt64 tillSlotInclusive, final int pruneLimit) {
     try (final Stream<DataColumnSlotAndIdentifier> prunableIdentifiers =
-            streamDataColumnIdentifiers(UInt64.ZERO, tillSlotInclusive);
+            streamDataColumnIdentifiers(UInt64.ZERO, tillSlotInclusive).limit(pruneLimit);
```

Comment:
you put limit on DataColumnIdentifiers while it's supposed to limit actual pruned slots assuming only slots containing sidecars. Take a look how limit is used in blob sidecars pruner